platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/maps/BitmapImageConversionTask.kt

```diff
@@ -0,0 +1,31 @@
+package org.maplibre.android.maps
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.maplibre.android.maps.Style.Builder.ImageWrapper
+import java.lang.ref.WeakReference
+
+internal class BitmapImageConversionTask(
+	nativeMap: NativeMap,
+) {
+	private val nativeMap = WeakReference(nativeMap)
+	private val job: Job = SupervisorJob(null)
+	private val scope = CoroutineScope(Dispatchers.IO + job)
+
+	fun execute(vararg params: ImageWrapper) {
+		scope.launch {
+			val images = params.map { Style.toImage(it) }
+			val nativeMap = nativeMap.get()
+			if (nativeMap != null && !nativeMap.isDestroyed) {
+				nativeMap.addImages(images.toTypedArray())
+			}
```

Comment:
Previously, this was done in the UI thread, now it is done in the IO thread. Is this fine?

Reply:
Most probably should stay UI

Reply:
Resolved.